tests/unit/test_storage.py

```diff
@@ -305,6 +305,14 @@ def test_get_ceph_versions_mismatch(self):
         inst.get_ceph_versions_mismatch()
         self.assertEqual(inst.output["ceph"]["versions"], result)
 
+    def test_get_ceph_mon_lower_version(self):
+        result = {'mgr': ['15.2.13'],
```

Comment:
This test checks whether `result` and the versions found in sos_commands/ceph/ceph_versions differ.
That would still be the case if sos_commands/ceph/ceph_versions contains the mismatched one as it's not different to `result`.

You'd want to test if `result` is identified as a problematic version set. You can instead use `CEPH_VERSIONS_MISMATCHED` for this test (modify the mon version there to lower one) and test whether both mismatches are identified (similar to `test_get_ceph_pg_imbalance`).

Reply:
One more thing. I found a bug in the  code that parses ceph versions output and have fixed it in https://github.com/canonical/hotsos/commit/5069fa2fb9647396dab6899d9aff82bc78ca4316. Please rebase your patch on top of master to pull that in (git rebase master).

Reply:
ack I will check on this